provides minimal troubleshooting for mrj (#3532)
reactor/reactor-core: README.md

}

dependencies {

>     compile "io.projectreactor:reactor-core:3.6.0-SNAPSHOT"

Reviewer: As the README.md will be displayed in the repo's GitHub entry point, I think we shouldn't advertise using the snapshot here.

Author: reverted

    testCompile "io.projectreactor:reactor-test:3.6.0-SNAPSHOT"

    // Alternatively, use the following for latest snapshot artifacts in this line

>     // compile "io.projectreactor:reactor-core:3.6.0-SNAPSHOT"

Reviewer: I think we can instead suggest "trying the upcoming release line" and use such wording.

Author: reverted

    // testCompile "io.projectreactor:reactor-test:3.5.8-SNAPSHOT"

    // Optionally, use `reactor-tools` to help debugging reactor code

>     // implementation "io.projectreactor:reactor-tools:3.5.8-SNAPSHOT"

Reviewer: this can probably also stay on the release version instead of snapshot

[complete note](https://projectreactor.io/docs/core/release/reference/docs/index.html#prerequisites)
in the reference guide.

## Trouble building the project?
Since the introduction of [Java Multi-Release JAR File](https://openjdk.org/jeps/238) 

> support one need to have JDK 8, 9, 21 available on the classpath. All the JDKs should 

Reviewer: ```suggestion
support one needs to have JDK 8, 9, and 21 available on the classpath. All the JDKs should 
```

be automatically [detected](https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection) 
or [provisioned](https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning) 
by Gradle Toolchain. 

However, if you see error message such as `No matching toolchains found for requested 
specification: {languageVersion=X, vendor=any, implementation=vendor-specific}` (where 

> `X` can be 8, 9 or 21), it means that you need to install missing JDK:

Reviewer: ```suggestion
`X` can be 8, 9 or 21), it means that you need to install the missing JDK:
```

### Installing JDKs with [SDKMAN!](https://sdkman.io/)

In the project root folder run [SDKMAN env initialization](https://sdkman.io/usage#env): 

```shell
sdk env install
```

then (if needed) install JDK 9: 

```shell
sdk install java 9.0.4-open
```

then (if needed) install JDK 21:

```shell
sdk install java 21.ea.28-open
```

> now you are done, try to refresh the project and see that it builds.

Reviewer: ```suggestion
When the installations succeed, try to refresh the project and see that it builds.
```

### Installing JDKs manually

The manual Operation-system specific JDK installation

> is well explained at the [official docs](https://docs.oracle.com/en/java/javase/20/install/overview-jdk-installation.html)

Reviewer: ```suggestion
is well explained in the [official docs](https://docs.oracle.com/en/java/javase/20/install/overview-jdk-installation.html)
```
